SimplifiedTests/NYPLAxisTests/NYPLPackageServiceTests/NYPLAxisPackageServiveTests.swift

```diff
@@ -0,0 +1,176 @@
+    }
+    itemDownloader.delegate = terminationListener
+    packageService.downloadPackageContent()
+    
+    waitForExpectations(timeout: 3, handler: nil)
+  }
+  
+  func testPackageServiceShouldNotContinueUponPackageWriteFailure() {
+    let terminationExpectation = self.expectation(
+      description: "Pacakge download should terminate if pacakge file write fails")
+    let writeExpectation = self.expectation(
+      description: "PackageService should try to write package file")
+    
+    let packageService = NYPLAxisPackageService(
+      axisItemDownloader: itemDownloader, axisKeysProvider: keysProvider,
+      baseURL: baseURL, parentDirectory: downloadsDirectory)
+    
+    mockDownloadedContainer()
+    contentDownloader.desiredResult = .success(packageData)
+    assetWriter.mockingFailure()
+    
+    let terminationListener = TerminationListenerMock()
+    terminationListener.didTerminate = {
+      terminationExpectation.fulfill()
+    }
+    
+    assetWriter.willWriteAsset = {
+      writeExpectation.fulfill()
+    }
+    
+    itemDownloader.delegate = terminationListener
+    packageService.downloadPackageContent()
+    
+    waitForExpectations(timeout: 3, handler: nil)
+  }
+  
+  func testPackageServiceShouldDownloadLinksFromPackageFile() {
+    let writeExpectation = self.expectation(
+      description: "PackageService should try to write package file")
+    
+    writeExpectation.expectedFulfillmentCount = 4
+    
+    let itemDownloader = NYPLAxisItemDownloader(
+      assetWriter: NYPLAssetWriter(), downloader: contentDownloader)
+    let packageService = NYPLAxisPackageService(
+      axisItemDownloader: itemDownloader, axisKeysProvider: keysProvider,
+      baseURL: baseURL, parentDirectory: downloadsDirectory)
+    
+    mockDownloadedContainer()
+    contentDownloader.desiredResult = .success(packageData)
+    
+    let terminationListener = TerminationListenerMock()
+    terminationListener.didTerminate = {
+      XCTFail()
+    }
+    
+    itemDownloader.delegate = terminationListener
+    packageService.downloadPackageContent()
+    let packageURL = downloadsDirectory.appendingPathComponent("ops/package.opf")
+    let chapter1URL = downloadsDirectory.appendingPathComponent("ops/xhtml/ch01.html")
+    let chapter2URL = downloadsDirectory.appendingPathComponent("ops/xhtml/ch02.html")
+    let chapter3URL = downloadsDirectory.appendingPathComponent("ops/xhtml/ch03.html")
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if self.fileExists(at: packageURL) {
+        writeExpectation.fulfill()
+      }
+      if self.fileExists(at: chapter1URL) {
+        writeExpectation.fulfill()
+      }
+      if self.fileExists(at: chapter2URL) {
+        writeExpectation.fulfill()
+      }
+      if self.fileExists(at: chapter3URL) {
+        writeExpectation.fulfill()
+      }
+    }
+    
+    waitForExpectations(timeout: 3, handler: nil)
+  }
+
+  private func mockDownloadedContainer() {
```

Comment:
I couldn't understand what this function was doing without looking at the code, can we rename this? Doe you think `writeAssetForContainer()` work here?

Reply:
Done